src/java/voldemort/client/protocol/RequestFormat.java

```diff
@@ -68,6 +70,17 @@ public void writeGetAllRequest(DataOutputStream output,
     public Map<ByteArray, List<Versioned<byte[]>>> readGetAllResponse(DataInputStream stream)
             throws IOException;
 
+    /*
+     * returns the max expected put request size, An implementation can avoid it
+     * by returning the value SIZE_UNKNOWN
+     */
+    public int getExpectedPutRequestSize(String storeName,
```

Comment:
I think you do this for only put because, thats the only place where we don't know if we are pulling out a rabbit or elephant from the storage engine..